src/dotnet/OpenAPI.Validator/Validation/XmsParameterLocation.cs

```diff
@@ -0,0 +1,60 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using AutoRest.Core.Logging;
+using OpenAPI.Validator.Properties;
+using OpenAPI.Validator.Validation.Core;
+using OpenAPI.Validator.Model;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace OpenAPI.Validator.Validation
+{
+    public class XmsParameterLocation : TypedRule<SwaggerParameter>
+    {
+        private static readonly IEnumerable<string> AllowedGlobalParameters = new List<string>()
+        { "subscriptionid", "api-version", "apiversion" };
+
+        /// <summary>
+        /// Id of the Rule.
+        /// </summary>
+        public override string Id => "R4001";
+
+        /// <summary>
+        /// Violation category of the Rule.
+        /// </summary>
+        public override ValidationCategory ValidationCategory => ValidationCategory.SDKViolation;
+
+        /// <summary>
+        /// The template message for this Rule. 
+        /// </summary>
+        /// <remarks>
+        /// This may contain placeholders '{0}' for parameterized messages.
+        /// </remarks>
+        public override string MessageTemplate => Resources.XmsParameterLocation;
+
+        /// <summary>
+        /// The severity of this message (ie, debug/info/warning/error/fatal, etc)
+        /// </summary>
+        public override Category Severity => Category.Warning;
```

Comment:
Should this be an error to get the author's attention. No one looks at warnings and we would probably miss it and catch much later. If they actually need this then the error can always be suppressed. Based on my experience 99% of services for Management plane should not have any parameters apart from subscriptionId, and api-version in global parameters section without `"x-ms-parameter-location": "method"`